biz.aQute.bndlib/src/aQute/bnd/service/reporter/ReportSerializerPlugin.java

```diff
@@ -0,0 +1,31 @@
+package aQute.bnd.service.reporter;
+
+import java.io.OutputStream;
+import java.util.Map;
+
+import org.osgi.annotation.versioning.ProviderType;
+
+/**
+ * This plugin serialize the extracted metadata DTO into a specific format.
+ */
+@ProviderType
+public interface ReportSerializerPlugin {
+
+	/**
+	 * Get the set of file extension names corresponding to the format that this
+	 * plugin can serialize into.
+	 * 
+	 * @return one or multiple extensions name, never {@code null}
+	 */
+	public String[] getHandledExtensions();
+
+	/**
+	 * Serialize the metadata DTO into the output stream.
+	 * 
+	 * @param metadata the metadata to serialize, must not be {@code null}
+	 * @param output the output stream to write the serialization result, must
+	 *            not be {@code null}
+	 * @throws Exception if any errors occur during the serialization process
+	 */
+	public void serialize(Map<String,Object> metadata, OutputStream output) throws Exception;
```

Comment:
Why isnt this

```java
public void serialize(Object dto, OutputStream output);
```
 Since this is about serializing a DTO-like object and not just a map?

Reply:
Since the plugin can handle multiple formats, what format is the serialize method to generate? Perhaps the requested format should be a parameter to the serialize method?

Reply:
> Since this is about serializing a DTO-like object and not just a map?

The metadata root is of type Map<String,Object>, this plugin is here to serialize the report in this context, it is not a general plugin that serializes DTO.

>Since the plugin can handle multiple formats, what format is the serialize method to generate? Perhaps the requested format should be a parameter to the serialize method?

The provider implements a specific serialization format and provide one or multiple standard extension names for that format (with the `getHandledExtensions`, perhaps this would be component properties in DS). Multiple implementations of this plugin are collected by the report generator processor and based on user configuration, it selects the good implementation (which matches the desired format).

Reply:
If the serializer takes a map as input, why doesn't the deserializer produce a map as output? Seems a symmetry problem.

Reply:
The deserializer does not produce a report but a DTO representation of a file that is added to the report (see comments ReportImportDeserializer).